pkg/menu/menu.go

```diff
@@ -49,25 +81,120 @@ func processInput(input, warning *widgets.Paragraph, isValid func(string) (strin
 				warning.Text = ""
 				ui.Render(warning)
 			}
-			input.Text += e.ID
+			input.Text += k
 			ui.Render(input)
 		}
 	}
 }
 
-// GetInput will present an input box to user and return the user's input.
-// GetInput will check validation of input using isValid function.
-func GetInput(introwords string, location int, wid int, ht int, isValid func(string) (string, bool)) (string, error) {
-	intro := newParagraph(introwords, false, location, len(introwords)+4, 3)
-	location += 2
-	input := newParagraph("", true, location, wid, ht+2)
-	location += ht + 2
-	warning := newParagraph("", false, location, wid, 3)
+// NewCustomInputWindow creates a new ui window and displays an input box.
+func NewCustomInputWindow(introwords string, wid int, ht int, isValid func(string) (string, string, bool)) (string, error) {
+	uiEvents := ui.PollEvents()
+	inputString, err := newInputWindow(introwords, wid, ht, isValid, uiEvents)
```

Comment:
You can just directly do `return newInputWindow(...)` since the return type is the same! Same for below